### PR TITLE
feat: Improve optional chaining validation

### PR DESCRIPTION
**Pull request by sunrabbit123**
**Description:**

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

File: crates/stc_ts_type_checker/tests/conformance/controlFlow/controlFlowOptionalChain.error-diff.json [hunk 1]
```diff
@@ -25,6 +25,8 @@
       223,
       241,
       244,
+      370,
```
Comment (sunrabbit123) [inline crates/stc_ts_type_checker/tests/conformance/controlFlow/controlFlowOptionalChain.error-diff.json:28]: else, there should be separate logic in there, but putting it in false fact doesn't seem to work.
What am I missing?

Comment (sunrabbit123) [reply crates/stc_ts_type_checker/tests/conformance/controlFlow/controlFlowOptionalChain.error-diff.json:28]:
```ts
type Thing = { foo: string | number, bar(): number, baz: object };

function f16(o: Thing | undefined) {
    if (o?.foo === undefined) {
        o.foo;  // Error
    }
    else {
        o.foo;
    }
    if (o?.foo !== undefined) {
        o.foo;
    }
    else {
        o.foo;  // L370
    }
    if (o?.foo == undefined) {
        o.foo;  // Error
    }
    else {
        o.foo;
    }
    if (o?.foo != undefined) {
        o.foo;
    }
    else {
        o.foo;  // L382
    }
}
```

**Review by kdy1 — APPROVED** [review]
Thank you! Can you rebase?

**Comment by sunrabbit123** [comment]
sir, i rebased it

**Review by kdy1 — APPROVED** [review]

**Pull request merged**